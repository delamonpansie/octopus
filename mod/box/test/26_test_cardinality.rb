#!/usr/bin/ruby

$: << File.dirname($0) + '/lib'
require 'run_env'

class Env < RunEnv
  def config
    super + <<EOD
object_space[0].cardinality = 2
EOD
  end
end

Env.connect_eval do
  insert ['a', 'b']
  log_try { insert ['a'] }
  log_try { insert ['a', 'b', 'c'] }

  select 'a'
  log_try { update_fields 'a', [1, :delete] }
  update_fields 'a', [1, :delete], [1, :insert, 'c']
  select 'a'
end
