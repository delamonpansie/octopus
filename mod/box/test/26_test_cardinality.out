# box.insert(["a", "b"])
1

# box.insert(["a"])
Failed with: {code: 0x202, message: 'tuple cardinality must match object_space cardinality'}
# box.insert(["a", "b", "c"])
Failed with: {code: 0x202, message: 'tuple cardinality must match object_space cardinality'}
# box.select("a")
[["a", "b"]]

# box.update_fields("a", [1, :delete])
Failed with: {code: 0x202, message: 'tuple cardinality must match object_space cardinality'}
# box.update_fields("a", [1, :delete], [1, :insert, "c"])
1

# box.select("a")
[["a", "c"]]

